src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -87,130 +97,139 @@ public function handle(AddProductToOrderCommand $command)
+            $this->assertOrderWasNotShipped($order);
 
-        $cart = $this->createNewCart($order);
+            $product = $this->getProductObject($command->getProductId(), (int) $order->id_lang);
+            $combination = $this->getCombination($command->getCombinationId());
 
-        $specificPrice = $this->createSpecificPriceIfNeeded(
-            $command,
-            $order,
-            $cart,
-            $product,
-            $combination
-        );
+            $cart = $this->createNewCart($order);
 
-        $this->addProductToCart($cart, $product, $combination, $command->getProductQuantity());
+            $specificPrice = $this->createSpecificPriceIfNeeded(
+                $command,
+                $order,
+                $cart,
+                $product,
+                $combination
+            );
 
-        $invoice = $this->createNewOrEditExistingInvoice(
-            $command,
-            $order,
-            $cart
-        );
+            $this->addProductToCart($cart, $product, $combination, $command->getProductQuantity());
 
-        $totalMethod = $command->getOrderInvoiceId() ? Cart::BOTH_WITHOUT_SHIPPING : Cart::BOTH;
+            $invoice = $this->createNewOrEditExistingInvoice(
+                $command,
+                $order,
+                $cart
+            );
 
-        // Create Order detail information
-        $orderDetail = new OrderDetail();
-        $orderDetail->createList(
-            $order,
-            $cart,
-            $order->getCurrentOrderState(),
-            $cart->getProducts(),
-            !empty($invoice->id) ? $invoice->id : 0
-        );
+            $totalMethod = $command->getOrderInvoiceId() ? Cart::BOTH_WITHOUT_SHIPPING : Cart::BOTH;
 
-        // update totals amount of order
-        // @todo: use https://github.com/PrestaShop/decimal for prices computations
-        $order->total_products += (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
-        $order->total_products_wt += (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS);
+            // Create Order detail information
+            $orderDetail = new OrderDetail();
+            $orderDetail->createList(
+                $order,
+                $cart,
+                $order->getCurrentOrderState(),
+                $cart->getProducts(),
+                !empty($invoice->id) ? $invoice->id : 0
+            );
 
-        $order->total_paid += Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
-        $order->total_paid_tax_excl += Tools::ps_round((float) $cart->getOrderTotal(false, $totalMethod), 2);
-        $order->total_paid_tax_incl += Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
+            // update totals amount of order
+            // @todo: use https://github.com/PrestaShop/decimal for prices computations
+            $order->total_products += (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
```

Comment:
For the `total_products_wt` I noticed a bug #17976 but there are some other prices that are not rounded. I don't want to round them all and introduce myself a bug

Maybe an advice from @eternoendless @matks @jolelievre ... @PrestaShop/prestashop-core-developers  ?

Reply:
This is very weird, and seems like a quick fix that blow up later.. Why round only with taxes? This may work with the values you tried but t doesn't it'll work in every cases
I know it's a very difficult topic, and I'd say it needs to dealt with in a dedicated PR, and in a transversal way in all handlers Not just this one, and just on this value..

Reply:
If I understand well, you mean that fixing that may pop other bugs in other cases ?

And we need to refactor the rounding process everywhere at once ?

Reply:
@jolelievre Ping